client/src/components/HistoricalComponents/historicalData/addSessionModal.js

```diff
@@ -1,97 +1,82 @@
-import React from "react";
+import React, { useState } from "react";
 import { Row, Col, Table, Button, Form, Modal } from "react-bootstrap";
+import Select from 'react-select';
 import modalStyling from './_styling/modalStyling.module.css';
 
+const subTeamOptions = [
+  { value: 1, label: 'Frame & Body' },
+  { value: 2, label: 'Suspension' },
+  { value: 3, label: 'Powertrain' },
+  { value: 4, label: 'Electrical' },
+  { value: 5, label: 'Aero' },
+];
 export default class AddSessionModal extends React.Component {
   constructor(props) {
     super(props);
     this.name = React.createRef();
-    this.subteam = React.createRef();
-
     this.state = {
       show: this.props.show,
       hide: this.props.hide,
+      subteam: [],
     };
-    this.forms = [];
-    this.formData = [];
   }
 
   componentDidMount = () => {
-    this.generateForms();
   };
 
   submit = async () => {
-    let name = this.name.current.value;
-    let subteam = this.name.current.value;
+    if(this.name.current.value !== '' && this.state.subteam.length !== 0)
+    {
+      let name = this.name.current.value;
+      let subteam = this.state.subteam.join();
 
-    this.props.submit(name, subteam);
-    this.props.hide();
+      this.props.submit(name, subteam);
+      this.props.hide();
+    }
   };
 
-  setData = (ref) => {
-    this.formData.push(ref);
+
+  onChangeSubteam = (selectedOption) => {
+    this.state.subteam = [];
```

Comment:
So changing the state directly like that will not rerender the component., you gotta use `this.setState()`.